app/services/service_provider_session_creator.rb

```diff
@@ -8,14 +8,14 @@ def initialize(sp:, view_context:, sp_session:, service_provider_request:)
 
   def call
     if sp
-      ServiceProviderSessionDecorator.new(
```

Comment:
as long as we're here, could we consider renaming this from `#call` to `#create` or maybe even `#session`? Makes it clearer what the result is IMO

Reply:
That bugged me too. I went for `create_session` since we have so many uses of `session` already.